### PR TITLE
Adding more testing on Int

### PR DESCRIPTION
**Pull request by tudorcebere**
## Description
Added a sanity check that ids behave as we should expect on `Int`

## Affected Dependencies
None

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by Jasopaum — APPROVED** [review]
LGTM

File: src/syft/lib/python/primitive_factory.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 
 class PrimitiveFactory:
     @staticmethod
     @syft_decorator(typechecking=True)
     def generate_primitive(
-        value: Union[int, float, bool, complex, None]
+        value: Union[int, float, bool, complex, None], id: Optional[UID] = None
     ) -> PyPrimitive:
         from .int import Int
```
Comment (gmuraru) [inline src/syft/lib/python/primitive_factory.py:14]: Q: Can't we move those outside?

File: src/syft/lib/python/primitive_interface.py [hunk 2]
```diff
@@ -1,2 +1,3 @@
 class PyPrimitive:
```
Comment (gmuraru) [inline src/syft/lib/python/primitive_interface.py:1]: Q: Abstract Class?

File: src/syft/lib/python/primitive_factory.py (hunk 1 above)
Comment (iamtrask) [reply src/syft/lib/python/primitive_factory.py:14]: It would be great to move these imports before we merge.

Comment (tudorcebere) [reply src/syft/lib/python/primitive_factory.py:14]: It is a circular dependency, let it be for now and fix it when we have it all.

File: src/syft/lib/python/primitive_interface.py (hunk 2 above)
Comment (tudorcebere) [reply src/syft/lib/python/primitive_interface.py:1]: :+1:

File: src/syft/lib/python/primitive_factory.py (hunk 1 above)
Comment (iamtrask) [reply src/syft/lib/python/primitive_factory.py]: Mmm - in my experience circular dependencies only get worse the longer you wait. We are under a deadline so I'm tempted this time - but technical_debt+=1

**Pull request merged**